Source/Dafny/Verifier/Translator.cs

```diff
@@ -7919,6 +7924,34 @@ enum MethodTranslationKind { SpecWellformedness, Call, CoCall, Implementation, O
       }
     }
 
+    static string MethodVerboseName(string fullName, MethodTranslationKind kind) {
+      Contract.Requires(fullName != null);
+      switch (kind) {
+        case MethodTranslationKind.SpecWellformedness:
+          return fullName + " (well-formedness)";
+        case MethodTranslationKind.Call:
+          return fullName + " (call)";
+        case MethodTranslationKind.CoCall:
+          return fullName + " (co-call)";
+        case MethodTranslationKind.Implementation:
+          return fullName + " (correctness)";
+        case MethodTranslationKind.OverrideCheck:
+          return fullName + " (override check)";
+        default:
+          Contract.Assert(false);  // unexpected kind
+          throw new cce.UnreachableException();
+      }
+    }
+
+    private static void AddVerboseName(Bpl.NamedDeclaration boogieDecl, string dafnyName, MethodTranslationKind kind) {
+      var name = MethodVerboseName(dafnyName, kind);
+      boogieDecl.AddAttribute("verboseName", new object[] { name });
+    }
+
+    private static void CopyVerboseName(Bpl.NamedDeclaration targetDecl, Bpl.NamedDeclaration sourceDecl) {
+      targetDecl.AddAttribute("verboseName", new object[] { sourceDecl.VerboseName });
+    }
```

Comment:
Use AddVerboseName here?  After that I would think you could even directly inline the call.

Reply:
I don't think that'd work, because `AddVerboseName` calls `MethodVerboseName`, which we don't want to do here. But I think there is a refactoring that could make sense here. I'll do it and you can let me know what you think.